pandas/core/nanops.py

```diff
@@ -1299,9 +1302,7 @@ def _ensure_numeric(x):
                 x = complex(x)
             except ValueError:
                 # e.g. "foo"
-                raise TypeError(
-                    "Could not convert {value!s} to numeric".format(value=x)
-                )
+                raise TypeError(f"Could not convert {x!s} to numeric")
```

Comment:
no need for ``!s``, you an use ``repr(x)`` if you want

Reply:
same, ok to change here

Reply:
I changed `x!s` to `repr(x)` but now a test fails. The reason is that the test expects `"foo"` whereas `repr` returns `"'foo'"`. Should I replace `repr` with `str` or get rid of `repr`?

Reply:
use repr where it makes sense
if i have to fix a test that is ok

Reply:
The only failing test is related to this. Shall I update the test also?